### ISSUE TITLE
Feature: Installation and update

### ISSUE THREAD

**Issue by rubiin** [kRHYME7/Hyde-cli#47]
Make readme more user friendly. May include an optional video tour guide on install.
Also add sections on how to update to latest hyprdots

**Comment by kRHYME7** [comment]
Soon will change the goal of this project. At first, it only acts as helper and workarounds, but I guess we should consider turning this project as a wrapper for Hyde.

**Comment by rubiin** [comment]
I think we should make the cli as preferred option for installing dots in the readme . And the git clone as manual

**Comment by kRHYME7** [comment]
Not yet, for now. As the CLI is dependent on some install scripts from Hyde.
 I am now considering  an independent one in case we can't follow the  breaking changes upstream. 

Expect a markdown for these soon in README. 

` Hyde-tui ` 
Done: 
+ shell selection
+  patching themes
+ Sddm theme selectiom
+ Restoring configs


not yet:
+ nvidia handling ( will steal it from upstream or I'm considering forking the endeavorOS  ` nvidia-inst  ` pkg  )
Can you check if it's legal? I already tested it on my local and it works. I'm hesitant because no one ported this packaeg to AUR.  (nvidia-inst AUR is ready too, in my local machine 😄)
+ GRUB, (will copy upstream lol)


What's on my plate:
+ Hyde-TUI templating (for declarative formats for devs -  just bash lol ) 
+ Package installation 

I made a fast canvas:
1.  Everything works now my only issue is I really want to store the package array inside a variable not a file so that it will be faster than writing multiple files. (Just my ego and proof of concept) 
2. We can write it all in memory,  drawbacks are We should execute everything in a single shell , or do some exports. 

> Will try 2 this week if it's too much then will use the first method. 
 
![image](https://github.com/kRHYME7/Hyde-cli/assets/53417443/153b8420-a22b-4019-9e66-66172da39d9e)

**Comment by rubiin** [comment]
Also, I think we should rename the cli from "Hyde" to "hyde" , most clis have small case naming patterns

**Comment by rubiin** [comment]
Also another thing, the make version of the install requires you to `sudo`. Is it okay?

**Comment by kRHYME7** [comment]
about the Command being Uppercase, It's for clarity and being conscious of what the user types in the Terminal.

**Comment by rubiin** [comment]
nah not necessary but a typical linux person would definitely type `hyde` after installing as the aur name is also `hyde-cli-git` .

**Comment by kRHYME7** [comment]
Will leave it like that for now. 

My convention for this is like 

1st: Main Command
2nd: Actual function
3rd: [args/params]

For me this is too long, so I decided to make the first command stand out, indicating that this is the main Shell. Something like that. Will definitely make it lowercase if 2 or more requested it. ( inspired byt the Compositor command ` Hyprland ` )

**Comment by rubiin** [comment]
that makes sense. lets get on with the remaining milestones . I could help out on my free time with some issues

**Comment by kRHYME7** [comment]
I'm just annoyed with the Hyde.usage haha. 

I'm thinking of breaking it down to single file for each funtions. eg ` upgrade ` ` restore `  etc 

then will make a script to merge them later on, plus compile with complgen. It is not a perfect solution 

 But will make it easier for me and other contributors to find the function via file, then add its related function. 
 
 Another thing can we use endeavorOS nvidia-inst?

**Comment by kRHYME7** [comment]
@rubiin  Hello! the ` Hyde.usage ` now is auto generated

**Comment by rubiin** [comment]
@kRHYME7 cool. Will try out. It was a pain to maintain the usage file as we are adding so many things .

**Comment by kRHYME7** [comment]
This is exclusive for --dev (contributors) 

so ` Hyde --dev ` select the gen completion then it will output the Hyde.usgae in the terminal (stdou). 

or just using ./complgen.sh

**Comment by rubiin** [comment]
I was thinking of adding `jp2a` dependency when installing from aur for seamless experience

**Comment by kRHYME7** [comment]
This is used to render images in ascii right?

**Comment by rubiin** [comment]
yup.

**Comment by rubiin** [comment]
> I was thinking of adding `jp2a` dependency when installing from aur for seamless experience

@kRHYME7 is this there? Or should I make a pr.

**Comment by kRHYME7** [comment]
This will opt to install when running anything that renders images and not using kitty

**Comment by rubiin** [comment]
closing this as it looks covered

**Comment by rubiin** [comment]
@kRHYME7 you can take code plus inspirations for features from here https://github.com/JaKooLit/Hyprland-Dots that will make hyde a lil better

**Comment by rubiin** [comment]
@kRHYME7 how about a powersaver mode?
Like powersaver on would quicky turn off all these

[-a] [--animations]             Default animations
[-ba] [--borderangle]           Default border angle animation
[-b] [--blur]                   Default blur
[-o] [--opacity]                Default tranparency



and again on toggle would enable them

**Comment by kRHYME7** [comment]
Yes that would be helpful! But take note this is NOT set using ` hyprctl keywords ` so it will be a persistent mode across reboots.

**Comment by rubiin** [comment]
yeah lets make toggle act as what the name would suggest. I dont know what toggle does now atm

**Comment by kRHYME7** [comment]
> yeah lets make toggle act as what the name would suggest. I dont know what toggle does now atm

Toggle will unset/set everything being declared set. 

So when you run ` Hyde power save --[multiple_params] you could unset it using "toggle". Then you can set it back to set it again. I will be adding ` Hyde power save --all `  to set into power save all the [params]

**Comment by rubiin** [comment]
> > yeah lets make toggle act as what the name would suggest. I dont know what toggle does now atm
> 
> Toggle will unset/set everything being declared set.
> 
> So when you run `Hyde power save --[multiple_params] you could unset it using "toggle". Then you can set it back to set it again. I will be adding` Hyde power save --all ` to set into power save all the [params]

what will be the counterpart for save --all . If save --all would allow to restore the chnages prior to the command, then maybe I could add a waybar module with a button to turn on and off powersaver mode for convinience

**Comment by kRHYME7** [comment]
Is  ` Hyde power unset --all ` good? 

The `toggle` command actually just add/remove the sourced file of the ` Hyde power  options `  so it would still be a good Idea to have explicit on/off, in which will be handles by ` save --all ` and `unset --all`

**Comment by kRHYME7** [comment]
![image](https://github.com/user-attachments/assets/0c1b516c-5fa4-4232-987c-9eee62dd9ebc)

**Comment by rubiin** [comment]
yeah save on and save off sounds good.

**Comment by kRHYME7** [comment]
How I wish we could come up with a good plugin system for HyDE. It is easy to input files but managing it across updates and removing it is an extra work. 

This plugin system should be the backbone of the installer and anything HyDE extra  stuff.

**Comment by rubiin** [comment]
@kRHYME7 how about a command called `Hyde theme generate` that will generate bare bones themes according to Hyde sepcs. Would allow making themes easier.

Can you setup a sample theme repo? I would work on this

**Comment by kRHYME7** [comment]
Something ` Hyde theme export  --ouput_dir /path/to/theme_dir/ `?  Which will repack theme into a target directory.

> generate bare bones themes according to Hyde sepcs. Would allow making themes easier. 

Or this is a different thing?

**Comment by rubiin** [comment]
Nah . This is more like a skeleton theme.  `Hyde theme init` would just clone one of the themes branch from
https://github.com/prasanthrangan/hyde-themes and rename cloned folder as `new-theme`. Then the user can go to this dir and modify the colors, icons etc

**Comment by rubiin** [comment]
Additionally we can also add instructions after the init from this discussions for further steps

https://github.com/prasanthrangan/hyprdots/discussions/653

**Comment by kRHYME7** [comment]
Aha yes! This is actually a TODO after hyde makes a default theme (using that as base theme), but at this state the repo is stale. 

Also, I noticed that different themes can have different approach, would it be nice that the ` init ` command will just copy the current theme and will ask for the new theme name afterward and creates it?

**Comment by rubiin** [comment]
@kRHYME7 https://github.com/kRHYME7/Hyde-cli/blob/42a9b599dab16175f8734faafeac380f47f88562/Scripts/Launch-Calculator#L3C16-L3C20 the qalc package name is incorrect. correct package is https://archlinux.org/packages/extra/x86_64/libqalculate/

**Comment by kRHYME7** [comment]
did it cause errors?

**Comment by rubiin** [comment]
yes i can. Yeah when launching 
![image](https://github.com/user-attachments/assets/3bb2c6df-0e21-44f2-9c53-95c1122344ae)

**Comment by kRHYME7** [comment]
` Hyde update `  then ` Hyde restore Config `  Users needs to resolve the files/dirs.

**Comment by rubiin** [comment]
Confused on Hyde upgrade. What does it do. We should include on readme how users can remain in sync with the main repo

**Comment by rubiin** [comment]
Can do a PR shortly for this

**Comment by kRHYME7** [comment]
Yes please. 

` upgrade ` is use to upgrade everything like rerunning the ` Hyde-install` 

` restore ` is used to restore the dotfiles that are declared in the ./restore_cfg.lst 

![image](https://github.com/user-attachments/assets/089881e8-4cea-4a93-bf69-8026fb909cf9)

**Comment by kRHYME7** [comment]
@rubiin  Hi!, can you Dm me on Discord. hehe I got some proposal. Thank you!

**Comment by rubiin** [comment]
> @rubiin Hi!, can you Dm me on Discord. hehe I got some proposal. Thank you!

sure thing

**Comment by rubiin** [comment]
@kRHYME7 
![hyde (1)](https://github.com/user-attachments/assets/5bbdc23c-9a6b-417e-84aa-2719311524d7)
heres the high quality image

**Comment by rubiin** [comment]
Maybe we can also add an animation style changer on rofi seems, doable.  If rofi supports a gif display that is. Will be a cool addition.

**Comment by kRHYME7** [comment]
Can rofi display gifs?

**Comment by rubiin** [comment]
maybe. But we do not need to display gif. Simply put on readme the what the gifs look like in actual
